Add required steps for installing OSS Calico on Openshift v4.16+ on AWS

coutinhop opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][DOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

Product Version(s):
<!--- Specify which versions of Calico, Calico Enterprise, and Calico Cloud your PR applies to. -->

Issue:
<!--- Add a link to the Jira ticket or GitHub issue, if applicable. --->

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

SME review:
- [ ] An SME has approved this change. 
<!--- SME approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

DOCS review:
- [ ] A member of the docs team has approved this change.
<!-- The Docs team must review and approve all changes. -->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

Merge checklist:
<!--- Mandatory for docs team members before merging code --->
- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed 


<!--- After you open your PR, ask for review from docs team:
  For community authors: tag @tigera/docs to ask for a review when your PR is ready --->

coutinhop committed:
Add required steps for installing OSS Calico on Openshift v4.16+ on AWS

radTuti commented on calico/getting-started/kubernetes/openshift/installation.mdx:
```diff
@@ -68,6 +68,38 @@ Now generate the Kubernetes manifests using your configuration file:
 
+:::note
+
+If you are using OpenShift **v4.16 or newer** on **AWS**, at this point you need to edit the OpenShift cluster-api manifests
```
```suggestion
For OpenShift **v4.16 or newer** on **AWS**, configure AWS security groups to allow BGP, typha and IP-in-IP encapsulation traffic by editing the OpenShift cluster-api manifests
```

radTuti commented on calico_versioned_docs/version-3.28/getting-started/kubernetes/openshift/installation.mdx:
```diff
@@ -68,6 +68,38 @@ Now generate the Kubernetes manifests using your configuration file:
+:::note
+
+If you are using OpenShift **v4.16 or newer** on **AWS**, at this point you need to edit the OpenShift cluster-api manifests
+that configure AWS security groups to allow BGP, typha and IP-in-IP encapsulation traffic in your cluster.
```
```suggestion
For OpenShift **v4.16 or newer** on **AWS**, configure AWS security groups to allow BGP, typha and IP-in-IP encapsulation traffic by editing the OpenShift cluster-api manifests
```

ctauchen replied:
I also prefer one sentence per line. Much easier to review in git.

radTuti commented on calico_versioned_docs/version-3.29/getting-started/kubernetes/openshift/installation.mdx:
```diff
@@ -68,6 +68,38 @@ Now generate the Kubernetes manifests using your configuration file:
+:::note
+
+If you are using OpenShift **v4.16 or newer** on **AWS**, at this point you need to edit the OpenShift cluster-api manifests
+that configure AWS security groups to allow BGP, typha and IP-in-IP encapsulation traffic in your cluster.
```
```suggestion
For OpenShift **v4.16 or newer** on **AWS**, configure AWS security groups to allow BGP, typha and IP-in-IP encapsulation traffic by editing the OpenShift cluster-api manifests
```

radTuti commented on calico/getting-started/kubernetes/openshift/installation.mdx:
````diff
@@ -68,6 +68,38 @@ Now generate the Kubernetes manifests using your configuration file:
+
+```
+yq -y -i '.spec.network.cni.cniIngressRules += [{ "description": "BGP (calico)", "fromPort": 179, "protocol": "tcp", "toPort": 179 }, { "description": "IP-in-IP (calico)", "fromPort": -1, "protocol": "4", "toPort": -1 }, { "description": "Typha (calico)", "fromPort": 5473, "protocol": "tcp", "toPort": 5473 }]' cluster-api/02_infra-cluster.yaml
+```
````
`yq` commands tends to be heavly dependant on the vers so perharps we do not want to include `yq` here
```suggestion
```

ctauchen replied:
I agree. Better to present one way to complete a task.

ctauchen reviewed (commented):
LGTM, with suggestions.

@coutinhop This PR covers OSS 3.28+. Am I right to assume this does not apply to 3.27? What about CE?

coutinhop committed:
Update calico_versioned_docs/version-3.29/getting-started/kubernetes/openshift/installation.mdx

Co-authored-by: tuti. <radTuti@live.ca>

coutinhop committed:
Address review comments

coutinhop commented:
> LGTM, with suggestions.
> 
> @coutinhop This PR covers OSS 3.28+. Am I right to assume this does not apply to 3.27? What about CE?

Thanks @ctauchen and @radTuti ! I've addressed your comments. Yes, this doesn't apply to v3.27 as the product changes to support OCP v4.16+ were only backported to v3.28+. The changes for Calico EE are still in flight, I'll change the docs once the product changes are merged.

ctauchen commented:
LGTM
